Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp

```diff
@@ -30,22 +30,37 @@ using namespace Spire::Styles;
 UiProfile Spire::make_box_profile() {
   auto properties = std::vector<std::shared_ptr<UiProperty>>();
   populate_widget_properties(properties);
+  properties.push_back(make_standard_int_property("border-size", 1));
+  properties.push_back(make_standard_int_property("border-radius", 0));
   properties.push_back(make_standard_bool_property("display_warning"));
   auto profile = UiProfile(QString::fromUtf8("Box"), properties,
     [] (auto& profile) {
       auto box = new Box(nullptr);
       box->resize(scale(100, 100));
-      auto style = StyleSheet();
-      style.get(Any()).
-        set(BackgroundColor(QColor::fromRgb(255, 255, 255))).
-        set(border(scale_width(1), QColor::fromRgb(0xC8, 0xC8, 0xC8))).
-        set(horizontal_padding(scale_width(8)));
-      style.get(Hover() || Focus()).
-        set(border_color(QColor::fromRgb(0x4B, 0x23, 0xA0)));
-      style.get(Disabled()).
-        set(BackgroundColor(QColor::fromRgb(0xF5, 0xF5, 0xF5))).
-        set(border_color(QColor::fromRgb(0xC8, 0xC8, 0xC8)));
-      set_style(*box, std::move(style));
+      auto& border_size = get<int>("border-size", profile.get_properties());
+      auto& border_radius = get<int>("border-radius",
+        profile.get_properties());
+      auto update_style = [&, box = box] {
+        auto style = StyleSheet();
+        style.get(Any()).
+          set(BackgroundColor(QColor::fromRgb(255, 255, 255))).
```

Comment:
This method of updating the stylesheet defeats the purpose of having a composable styling system. Only update the specific styling necessary to apply the change.